c7n/resources/ec2.py

```diff
@@ -302,6 +302,56 @@ def __call__(self, i):
         return self.operator(map(self.match, volumes))
 
 
+@filters.register('termination-protected')
+class DisableApiTermination(Filter):
+    """EC2 instances with ``disableApiTermination`` attribute set
+
+    Filters EC2 instances with ``disableApiTermination`` attribute set to true.
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: termination-protection-enabled
+            resource: ec2
+            filters:
+              - type: termination-protected
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: termination-protection-NOT-enabled
+            resource: ec2
+            filters:
+              - not:
+                - type: termination-protected
+    """
+
+    schema = type_schema('termination-protected')
+    permissions = ['ec2:DescribeInstanceAttribute']
+
+    def get_permissions(self):
+        perms = self.permissions
```

Comment:
so to keep from extending the mutable class variable this needs to be turned into a new list reference, via 
`perms = list(self.permissions)`